src/gobapi/dump/to_db.py

```diff
@@ -246,6 +246,90 @@ def dump_to_db(self, full_dump=False):
         yield from self._rename_tmp_table()
         yield from self._create_indexes(model)
 
+    def create_utility_view(self):
+        """Creates view with utility columns for relating without relation table
+
+        View contains all columns from the main table, plus the RELATION_id, RELATION_VOLGNUMMER, RELATION_ref
+        and RELATION_bronwaarde columns for each RELATION (for example ligt_in_buurt_id, ligt_in_buurt_ref and
+        ligt_in_buurt_bronwaarde)
+
+        :return:
+        """
+
+        main_alias = self.model['abbreviation'].lower()
+        src_has_states = self.model.get('has_states', False)
+
+        # Collect all necessary joins and select statements
+        joins = []
+        selects = [f'{main_alias}.*']
+
+        def ref(rel_alias: str, dst_has_states: bool):
+            return f"{rel_alias}.dst_id || '_' || {rel_alias}.dst_volgnummer" \
+                if dst_has_states else f"{rel_alias}.dst_id"
+
+        for relation in [k for k in self.model['references'].keys()]:
+            # Add a join and selects for each relation
+            relation_name = get_relation_name(GOBModel(), self.catalog_name, self.collection_name, relation)
+
+            if not relation_name:
+                pass
+
+            # Determine if ManyReference and if destination has states
+            src_field = self.model['all_fields'].get(relation)
+            dst_catalog_name, dst_collection_name = src_field['ref'].split(':')
+            dst_has_states = GOBModel().has_states(dst_catalog_name, dst_collection_name)
+            is_many = src_field['type'] == 'GOB.ManyReference'
+
+            on = f'{relation}.src_id = {main_alias}.{FIELD.ID}' + (
+                f' and {relation}.src_volgnummer = {main_alias}.{FIELD.SEQNR}' if src_has_states else ''
+            )
+
+            if is_many:
+                # For a ManyReference, we need to aggregate the values in an array
+                join = f"""
+left join (
+    select
+        rel.src_id,
+        array_agg(rel.dst_id) dst_id,
+        {'rel.dst_volgnummer dst_volgnummer,' if dst_has_states else ''}
+        array_agg(rel.bronwaarde) bronwaarde,
+        array_agg({ref('rel', dst_has_states)}) "ref"
+    from rel_{relation_name} rel
+    where rel.{FIELD.DATE_DELETED} is null
+    group by rel.src_id
+) {relation} on {on}
+"""
```

Comment:
I suggest putting comments in all SQL statements (where appropriate, like this one) in order to make them readable as a nicely document SQL statement

Reply:
👍